service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -1304,6 +1375,40 @@ private PaymentRequestDto formPayment(Long orderId, int sumToPay) {
         return paymentRequestDto;
     }
 
+    private int formCertificatesToBeSavedAndCalculateOrderSumClient(OrderFondyClientDto dto,
+        Set<Certificate> orderCertificates,
+        Order order, int sumToPay) {
+        if (sumToPay != 0 && dto.getCertificates() != null) {
+            boolean tooManyCertificates = false;
+            for (String temp : dto.getCertificates()) {
+                if (tooManyCertificates) {
+                    throw new TooManyCertificatesEntered(TOO_MANY_CERTIFICATES);
+                }
+                Certificate certificate = certificateRepository.findById(temp).orElseThrow(
+                    () -> new CertificateNotFoundException(CERTIFICATE_NOT_FOUND_BY_CODE + temp));
+                validateCertificate(certificate);
+                certificate.setOrder(order);
+                orderCertificates.add(certificate);
+                sumToPay -= certificate.getPoints();
+                certificate.setCertificateStatus(CertificateStatus.USED);
+                certificate.setDateOfUse(LocalDate.now());
+                if (dontSendLinkToFondyIfClient(sumToPay, certificate, dto)) {
+                    sumToPay = 0;
+                    tooManyCertificates = true;
+                }
+            }
+        }
+        return sumToPay;
```

Comment:
overall, method formCertificatesToBeSavedAndCalculateOrderSumClient seems to be complicated. Think about some simplification.